# Differential testing
All the steps of differential expression analysis can be done with a single call of the function `DESeq()`.

```{r}
meta$Factors <- paste(meta$Genotype, meta$Treatment, sep = '_')
```

```{r}
dds_factors <- DESeqDataSetFromMatrix(countData = round(data[,-c(1)]), colData = meta, design = ~ Factors)
```

```{r}
dds_analysis <- DESeq(dds_factors)
```

```{r dispest, cache = TRUE, fig.width = 6, fig.height = 5.5, fig.align = "center"}
plotDispEsts(dds_analysis)
```


```{r}
contrast_pbs <- c("Factors", "KO_PBS", "Control_PBS")
contrast_lps <- c("Factors", "KO_LPS", "Control_LPS")
```


```{r}
res_unshrunken_pbs <- results(dds_analysis, contrast=contrast_pbs, alpha = 0.05)
res_unshrunken_lps <- results(dds_analysis, contrast=contrast_lps, alpha = 0.05)
```

```{r}
res_pbs <- lfcShrink(dds_analysis, contrast=contrast_pbs, res=res_unshrunken_pbs, type='normal')
res_lps <- lfcShrink(dds_analysis, contrast=contrast_lps, res=res_unshrunken_lps, type='normal')
```

```{r ma_shrunken, cache = TRUE, fig.width = 6, fig.height = 5.5, fig.align = "center"}
plotMA(res_pbs, ylim=c(-3,5))
```


```{r}
summary(res_pbs, alpha = 0.05)
```

```{r}
summary(res_lps, alpha = 0.05)
```

```{r}
padj.cutoff <- 0.05
lfc.cutoff <- 0.58
```


```{r}
res_pbs_tb <- res_pbs %>% data.frame() %>% rownames_to_column(var="gene") %>% as_tibble()
res_lps_tb <- res_lps %>% data.frame() %>% rownames_to_column(var="gene") %>% as_tibble()
```

```{r}
head(res_pbs_tb)
head(res_lps_tb)
```


```{r}
sig_pbs <- res_pbs_tb %>%dplyr::filter(padj < padj.cutoff & abs(log2FoldChange) >= lfc.cutoff)
sig_lps <- res_lps_tb %>% dplyr::filter(padj < padj.cutoff & abs(log2FoldChange) >= lfc.cutoff)
```


```{r}
nrow(sig_pbs)
nrow(sig_lps)
```
