# Differential testing

Before the differential expression analysis we will combine $Genotype$ and $Treatment$ factors into one column, to create four groups for the design formula:

```{r}
meta$Factors <- paste(meta$Genotype, meta$Treatment, sep = '_')
meta$Factors <- factor(meta$Factors, levels = c("Control_PBS", "Control_LPS", "KO_PBS", "KO_LPS"))
```

Now re-create the `DESeqDataSet` object, but this time with a design formula consisting of $~Factors$:

```{r}
dds_factors <- DESeqDataSetFromMatrix(countData = round(data[,-c(1)]), 
                                      colData = meta, 
                                      design = ~ Factors)
```

Here we have only one factor for design formula, consisting of four different levels (Control_PBS, Control_LPS, KO_PBS, KO_LPS). In terms of linear models, the specified formula can be expressed as

$$expression = \beta_0 + \beta_1 Factor_{Control\_LPS} + \beta_2 Factor_{KO\_PBS} + \beta_3 Factor_{KO\_LPS}$$
or in matrix notations: 

$$\,
\begin{pmatrix}
Y_1\\
Y_2\\
\vdots\\
Y_N
\end{pmatrix} = 
\begin{pmatrix}
1 &x_{11} & x_{12} & x_{13}\\
1 &x_{21} & x_{22} & x_{23}\\
\vdots\\
1 & x_{N1} & x_{N2} & x_{N3}
\end{pmatrix}
\begin{pmatrix}
\beta_0\\
\beta_1\\
\beta_2\\
\beta_3
\end{pmatrix} +
\begin{pmatrix}
\varepsilon_1\\
\varepsilon_2\\
\vdots\\
\varepsilon_N
\end{pmatrix}
$$

where $X_{N\times4}$ is the design matrix, $\mathbf{\beta} = (\beta_0, \beta_1, \beta_2, \beta_3)^T$ are estimated coefficients (log2 Fold Changes) and $\mathbf{\varepsilon} = (\varepsilon_1, \varepsilon_2, ..., \varepsilon_N)^T$ are residuals.


Let's check the model matrix in dds object:
```{r}
model.matrix(design(dds_factors), data = colData(dds_factors))
```


## DE analysis

All the steps of differential expression analysis can be done with a single call of the function `DESeq()`:

```{r}
dds_analysis <- DESeq(dds_factors)
```

Plot dispersion estimates:

```{r dispest, cache = TRUE, fig.width = 6, fig.height = 5.5, fig.align = "center"}
plotDispEsts(dds_analysis)
```

To tell DESeq2 which groups we wish to compare, we supply the contrasts we would like to make using the contrast argument.

```{r}
contrast_pbs <- c("Factors", "KO_PBS", "Control_PBS")
contrast_lps <- c("Factors", "KO_LPS", "Control_LPS")
```

To build our results table we will use the results() function.

```{r}
res_unshrunken_pbs <- results(dds_analysis, contrast=contrast_pbs, alpha = 0.05)
res_unshrunken_lps <- results(dds_analysis, contrast=contrast_lps, alpha = 0.05)
```

To generate more accurate log2 fold change estimates, DESeq2 allows for the shrinkage of the log2 fold changes estimates toward zero when the information for a gene is low.

```{r}
res_pbs <- lfcShrink(dds_analysis, 
                     contrast=contrast_pbs, 
                     res=res_unshrunken_pbs, 
                     type='normal')
res_lps <- lfcShrink(dds_analysis, 
                     contrast=contrast_lps, 
                     res=res_unshrunken_lps, 
                     type='normal')
```

Check the DE results for the first comparison with PBS-treated samples:
```{r}
knitr::kable(head(res_pbs))
```
What are the columns of this table?

* `baseMean` is the mean gene expresion across all the samples in the data

* `log2FoldChange` represents differences for the comparison of interest (estimated $\beta_i$ coefficient)

* `lfcSE`	is an uncertainty of $\beta$ estimate ($\sigma_{\beta_i}$), represented by standard error (SE) for the log2(Fold Change) 

* `stat` is Z-statistic, also can be expressed as $Z = \frac{\beta_i}{\sigma_{\beta_i}}$

* `pvalue` means the raw p-value, derived from Wald Z-test, where $Z \sim N(0, 1)$,
 used to reject or accept the $H_0: \beta_i = 0$

* `padj` is an adjusted p-value (FDR)


To illustrate the effect of log2 foldchange shrinkage we will use the MA plot. The MA plot shows the mean of the normalized counts versus the log2 foldchanges for all genes tested. The genes that are significantly DE are colored.

```{r ma_shrunken, cache = TRUE, fig.width = 6, fig.height = 5.5, fig.align = "center"}
plotMA(res_pbs, ylim=c(-3,5))
```

::: design
**Task 2**: Plot the unshrunken results and save the plot as pdf. Are there any differences between unshrunken and shrunken results? Provide a brief explanation (1 point).
:::

Check the results summary for PBS and LPS comparisons:

```{r}
summary(res_pbs, alpha = 0.05)
```

```{r}
summary(res_lps, alpha = 0.05)
```


## Extracting significant genes

Now we are ready to extract significant differentially expressed genes. We will define significant genes as those satisfying the \|FDR p-value \< 0.05\| and \|log2(Fold Change)\| \> 0.58:

```{r}
padj.cutoff <- 0.05
lfc.cutoff <- 0.58
```

Convert the results table into a tibble:

```{r}
res_pbs_tb <- res_pbs %>% 
              data.frame() %>% 
              rownames_to_column(var="gene") %>% 
              as_tibble()
res_lps_tb <- res_lps %>% 
              data.frame() %>% 
              rownames_to_column(var="gene") %>% 
              as_tibble()
```

Select significant genes:

```{r}
sig_pbs <- res_pbs_tb %>% 
           dplyr::filter(padj < padj.cutoff & abs(log2FoldChange) >= lfc.cutoff)
sig_lps <- res_lps_tb %>% 
           dplyr::filter(padj < padj.cutoff & abs(log2FoldChange) >= lfc.cutoff)
```

```{r}
knitr::kable(head(sig_pbs))
```

```{r}
knitr::kable(head(sig_lps))
```

Intriguingly, the number of significant genes is almost twice more in the comparison with LPS compared to PBS-treated data.

```{r}
nrow(sig_pbs)
nrow(sig_lps)
```

::: design
**Task 3**: Perform DE analysis comparing LPS and PBS profiles: `Control_LPS vs Control_PBS` and `KO_LPS vs KO_PBS`. How many DE genes (FDP P-value < 0.05 and |log2FC|> 0.58) did you obtain for these comparisons (1 point).
:::

