# De novo transcriptome assembly  

## Read filtering
The first step is to remove adapters and low-quality reads from the raw .fastq files. To do this, run TrimGalore tool:

```{bash, eval = FALSE}
trim_galore --cores 2 --paired --gzip --fastqc --fastqc_args "-t 2" --output_dir data_filtered data/<your_number>.R1.fq.gz data/<your_number>.R2.fq.gz
```

where `--fastqc` argument means that we also want to perform quality assessment of filtered reads via FastQC using two computational cores (as specified by `--fastqc_args "-t 2"`).

## Run trinity
This is the most crucial and time consuming step in the analysis. It might take several hours, therefore it make sense to launch Trinity in screen session.

Create your screen session:
```{bash, eval = FALSE}
screen -S <session name>
```

Run Trinity:

```{bash, eval = FALSE}
Trinity --seqType fq --SS_lib_type RF --max_memory 2G --left data_filtered/<your_number>.R1_val_1.fq.gz --right data_filtered/<your_number>.R2_val_2.fq.gz --CPU 2
```

where `--SS_lib_type RF` means that the library preparation was done with a first strand protocol (we know that from the original paper).

Once Trinity run is finished, `trinity_out_dir` folder will be created. Inside it you will find `Trinity.fasta` reconstructed assembly and temporary files and folders generated by program during all the stages of assembling.

Check the `Trinity.fasta` header:

```{bash, eval = FALSE}
head Trinity.fasta
```

The output should have the following structure:

```{bash, eval = FALSE}
>TRINITY_DN486518_c0_g1_i1 len=531 path=[0:0-530]
GGGGACGGCTATGCGGCCAGTGTATGAGGAGGAGGAAGGGGAGCAACAAGACGATCATCC
TGTGTTCAAGAACGGCGAACCTGATTACCAAGGCGAATACCTTTCGTGCTTTGAAGAGCC
CAACCTGCTGTACCGCATTTGCAACTCGGAAGACAAAACATGGGCGTTCTATAACGATAG
CCGCATCTACGAGATACATGTGCAGTTTAATTTCGGCAAGAACTCCAAGTTGCAGCCGCT
TGAAAACACCAGCATGTACACGAGGAACTCGGGTGAGCATGTTGCCAGAGTCATTGTATA
CCCTGGCGAAACGGAAATGTTTGTGCAGGGCACCGTTAACGGCTTCACAAGCAAGCTTCG
CGCGATCCCACTTTCTGAGGAGCACCACAAGTGCCGCCAGGCAATCACTTCCGCCATGAT
TGAGGAAGAGGCGAATCAGATTCGGCAGATTGTCGGGGACGCTGCAGACGCCGAGACCGC
ACTGAAGGCGTGCGTTGAGAACAACATTCCGTTCGTCGACCCCGAGTTCCC
```





