<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 3 Normalization and Quality control | Omics Data Analysis</title>
  <meta name="description" content="This is a collection of seminars and homeworks for the Omics Data Analysis course at Skoltech." />
  <meta name="generator" content="bookdown 0.37 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 3 Normalization and Quality control | Omics Data Analysis" />
  <meta property="og:type" content="book" />
  <meta property="og:image" content="/docs/images/course_logo_v2.png" />
  <meta property="og:description" content="This is a collection of seminars and homeworks for the Omics Data Analysis course at Skoltech." />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 3 Normalization and Quality control | Omics Data Analysis" />
  
  <meta name="twitter:description" content="This is a collection of seminars and homeworks for the Omics Data Analysis course at Skoltech." />
  <meta name="twitter:image" content="/docs/images/course_logo_v2.png" />

<meta name="author" content="Dmitrii Smirnov, Skolkovo Institute of Science and Technology" />


<meta name="date" content="2024-03-25" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="input-dataset.html"/>
<link rel="next" href="differential-testing.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<link href="libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet" />
<script src="libs/pagedtable-1.1/js/pagedtable.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="toc-logo"><a href="./"><img src="images/Logo.png"></a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#how-to-deal-with-r"><i class="fa fa-check"></i><b>1.1</b> How to deal with R</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#assignment-and-grading"><i class="fa fa-check"></i><b>1.2</b> Assignment and grading</a></li>
</ul></li>
<li class="part"><span><b>Lecture 4: DE analysis</b></span></li>
<li class="chapter" data-level="2" data-path="input-dataset.html"><a href="input-dataset.html"><i class="fa fa-check"></i><b>2</b> Input dataset</a>
<ul>
<li class="chapter" data-level="2.1" data-path="input-dataset.html"><a href="input-dataset.html#count-table-importing"><i class="fa fa-check"></i><b>2.1</b> Count table importing</a></li>
<li class="chapter" data-level="2.2" data-path="input-dataset.html"><a href="input-dataset.html#deseq2-object"><i class="fa fa-check"></i><b>2.2</b> DESeq2 object</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="cross.html"><a href="cross.html"><i class="fa fa-check"></i><b>3</b> Normalization and Quality control</a>
<ul>
<li class="chapter" data-level="3.1" data-path="cross.html"><a href="cross.html#normalization"><i class="fa fa-check"></i><b>3.1</b> Normalization</a></li>
<li class="chapter" data-level="3.2" data-path="cross.html"><a href="cross.html#quality-control"><i class="fa fa-check"></i><b>3.2</b> Quality control</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="cross.html"><a href="cross.html#principal-component-analysis-pca"><i class="fa fa-check"></i><b>3.2.1</b> Principal Component Analysis (PCA)</a></li>
<li class="chapter" data-level="3.2.2" data-path="cross.html"><a href="cross.html#correlation-heatmap"><i class="fa fa-check"></i><b>3.2.2</b> Correlation Heatmap</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="differential-testing.html"><a href="differential-testing.html"><i class="fa fa-check"></i><b>4</b> Differential testing</a>
<ul>
<li class="chapter" data-level="4.1" data-path="differential-testing.html"><a href="differential-testing.html#de-analysis"><i class="fa fa-check"></i><b>4.1</b> DE analysis</a></li>
<li class="chapter" data-level="4.2" data-path="differential-testing.html"><a href="differential-testing.html#extracting-significant-genes"><i class="fa fa-check"></i><b>4.2</b> Extracting significant genes</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="gene-annotation.html"><a href="gene-annotation.html"><i class="fa fa-check"></i><b>5</b> Gene annotation</a></li>
<li class="chapter" data-level="6" data-path="visualization.html"><a href="visualization.html"><i class="fa fa-check"></i><b>6</b> Visualization</a>
<ul>
<li class="chapter" data-level="6.1" data-path="visualization.html"><a href="visualization.html#volcano-plot"><i class="fa fa-check"></i><b>6.1</b> Volcano plot</a></li>
<li class="chapter" data-level="6.2" data-path="visualization.html"><a href="visualization.html#heatmap-of-significant-genes"><i class="fa fa-check"></i><b>6.2</b> Heatmap of significant genes</a></li>
<li class="chapter" data-level="6.3" data-path="visualization.html"><a href="visualization.html#comparison-with-dam-markers"><i class="fa fa-check"></i><b>6.3</b> Comparison with DAM markers</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="functional-analysis.html"><a href="functional-analysis.html"><i class="fa fa-check"></i><b>7</b> Functional analysis</a>
<ul>
<li class="chapter" data-level="7.1" data-path="functional-analysis.html"><a href="functional-analysis.html#gene-ontology-go-analysis"><i class="fa fa-check"></i><b>7.1</b> Gene Ontology (GO) analysis</a></li>
<li class="chapter" data-level="7.2" data-path="functional-analysis.html"><a href="functional-analysis.html#pathway-analysis"><i class="fa fa-check"></i><b>7.2</b> Pathway analysis</a></li>
<li class="chapter" data-level="7.3" data-path="functional-analysis.html"><a href="functional-analysis.html#gene-set-enrichment-analysis-gsea"><i class="fa fa-check"></i><b>7.3</b> Gene-Set Enrichment Analysis (GSEA)</a></li>
<li class="chapter" data-level="7.4" data-path="functional-analysis.html"><a href="functional-analysis.html#gene-set-enrichment-analysis-with-msigdb"><i class="fa fa-check"></i><b>7.4</b> Gene Set Enrichment Analysis with mSigDB</a></li>
</ul></li>
<li class="part"><span><b>Lecture 9: Transcriptome</b></span></li>
<li class="chapter" data-level="8" data-path="introduction-1.html"><a href="introduction-1.html"><i class="fa fa-check"></i><b>8</b> Introduction</a>
<ul>
<li class="chapter" data-level="8.1" data-path="introduction-1.html"><a href="introduction-1.html#assignment-and-grading-1"><i class="fa fa-check"></i><b>8.1</b> Assignment and grading</a></li>
<li class="chapter" data-level="8.2" data-path="introduction-1.html"><a href="introduction-1.html#set-up-conda-environment"><i class="fa fa-check"></i><b>8.2</b> Set up conda environment</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="de-novo-transcriptome-assembly.html"><a href="de-novo-transcriptome-assembly.html"><i class="fa fa-check"></i><b>9</b> De novo transcriptome assembly</a>
<ul>
<li class="chapter" data-level="9.1" data-path="de-novo-transcriptome-assembly.html"><a href="de-novo-transcriptome-assembly.html#read-filtering"><i class="fa fa-check"></i><b>9.1</b> Read filtering</a></li>
<li class="chapter" data-level="9.2" data-path="de-novo-transcriptome-assembly.html"><a href="de-novo-transcriptome-assembly.html#run-trinity"><i class="fa fa-check"></i><b>9.2</b> Run trinity</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="transcriptome-assembly-qc.html"><a href="transcriptome-assembly-qc.html"><i class="fa fa-check"></i><b>10</b> Transcriptome assembly QC</a>
<ul>
<li class="chapter" data-level="10.1" data-path="transcriptome-assembly-qc.html"><a href="transcriptome-assembly-qc.html#obtain-assembly-statistics"><i class="fa fa-check"></i><b>10.1</b> Obtain assembly statistics</a></li>
<li class="chapter" data-level="10.2" data-path="transcriptome-assembly-qc.html"><a href="transcriptome-assembly-qc.html#transcript-filtering"><i class="fa fa-check"></i><b>10.2</b> Transcript filtering</a></li>
<li class="chapter" data-level="10.3" data-path="transcriptome-assembly-qc.html"><a href="transcriptome-assembly-qc.html#assembly-completeness-with-gvolante"><i class="fa fa-check"></i><b>10.3</b> Assembly completeness with gVolante</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="quantification-and-annotation.html"><a href="quantification-and-annotation.html"><i class="fa fa-check"></i><b>11</b> Quantification and annotation</a>
<ul>
<li class="chapter" data-level="11.1" data-path="quantification-and-annotation.html"><a href="quantification-and-annotation.html#transcript-quantification"><i class="fa fa-check"></i><b>11.1</b> Transcript quantification</a></li>
<li class="chapter" data-level="11.2" data-path="quantification-and-annotation.html"><a href="quantification-and-annotation.html#transcript-annotation"><i class="fa fa-check"></i><b>11.2</b> Transcript annotation</a></li>
<li class="chapter" data-level="11.3" data-path="quantification-and-annotation.html"><a href="quantification-and-annotation.html#integrating-expression-data-and-annotation"><i class="fa fa-check"></i><b>11.3</b> Integrating expression data and annotation</a></li>
</ul></li>
<li class="part"><span><b>Lecture 13: Lipidomics</b></span></li>
<li class="chapter" data-level="12" data-path="lipidomics-data-analysis.html"><a href="lipidomics-data-analysis.html"><i class="fa fa-check"></i><b>12</b> Lipidomics Data Analysis</a></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="part"><span><b>Session info</b></span></li>
<li class="chapter" data-level="13" data-path="session-info.html"><a href="session-info.html"><i class="fa fa-check"></i><b>13</b> Session Info</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Omics Data Analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="cross" class="section level1 hasAnchor" number="3">
<h1><span class="header-section-number">Chapter 3</span> Normalization and Quality control<a href="cross.html#cross" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="normalization" class="section level2 hasAnchor" number="3.1">
<h2><span class="header-section-number">3.1</span> Normalization<a href="cross.html#normalization" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The next step is to normalize the count data in order to be able to make fair gene comparisons between samples.</p>
<p>To perform the median of ratios method of normalization<a href="#fn1" class="footnote-ref" id="fnref1"><sup>1</sup></a>, <code>DESeq2</code> has a single <code>estimateSizeFactors()</code> function that will generate size factors for us. We will use the function in the example below, but in a typical RNA-seq analysis this step is automatically performed by the <code>DESeq()</code> function, which we will see later.</p>
<p>Perform normalization:</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="cross.html#cb12-1" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">estimateSizeFactors</span>(dds)</span></code></pre></div>
<p>We can take a look at the normalization factor applied to each sample:</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="cross.html#cb13-1" tabindex="-1"></a><span class="fu">sizeFactors</span>(dds)</span></code></pre></div>
<pre><code>##      KO_PBS_642      KO_PBS_641      KO_PBS_638      KO_LPS_600      KO_LPS_194      KO_LPS_193 Control_PBS_175 Control_PBS_171 Control_PBS_170 Control_LPS_185 
##       1.2295376       0.7077297       1.1883411       0.9626383       0.7803893       1.1168247       1.3162828       1.0183880       1.0247178       0.9783637 
## Control_LPS_184 Control_LPS_149 
##       1.0500173       0.9531096</code></pre>
<p>Now, to retrieve the normalized counts matrix from dds, use the <code>counts()</code> function and add the argument <code>normalized=TRUE</code>.</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="cross.html#cb15-1" tabindex="-1"></a>normalized_counts <span class="ot">&lt;-</span> <span class="fu">counts</span>(dds, <span class="at">normalized=</span><span class="cn">TRUE</span>)</span></code></pre></div>
<p>These normalized counts are useful for the QC and visualization purposes, but the step with <code>estimateSizeFactors()</code> is done automatically by <code>DESeq2</code> during differential expression analysis.</p>
</div>
<div id="quality-control" class="section level2 hasAnchor" number="3.2">
<h2><span class="header-section-number">3.2</span> Quality control<a href="cross.html#quality-control" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Next, we perform QC checks on the count data to help us ensure that the samples/replicates look good.</p>
<p>To explore the similarity of our samples, we will be performing sample-level QC using Principal Component Analysis (PCA) and hierarchical clustering methods.</p>
<p>When using these unsupervised clustering methods, transformation of the normalized counts may improve the distances/clustering for visualization. First, let’s try the most simple method of log2 transformation and plot counts of the two first KO_PBS replicates as a scatter plot:</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="cross.html#cb16-1" tabindex="-1"></a>nts <span class="ot">&lt;-</span> <span class="fu">log2</span>(<span class="fu">assay</span>(dds, <span class="at">normalized =</span> <span class="cn">TRUE</span>)<span class="sc">+</span><span class="dv">1</span>)</span>
<span id="cb16-2"><a href="cross.html#cb16-2" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="fu">as.data.frame</span>(nts), <span class="fu">aes</span>(<span class="at">x =</span> KO_PBS_642, <span class="at">y =</span> KO_PBS_641)) <span class="sc">+</span> </span>
<span id="cb16-3"><a href="cross.html#cb16-3" tabindex="-1"></a>    <span class="fu">geom_point</span>()<span class="sc">+</span></span>
<span id="cb16-4"><a href="cross.html#cb16-4" tabindex="-1"></a>    <span class="fu">ggtitle</span>(<span class="st">&#39;log2(x+1) transformation&#39;</span>)<span class="sc">+</span></span>
<span id="cb16-5"><a href="cross.html#cb16-5" tabindex="-1"></a>    <span class="fu">theme_dose</span>(<span class="dv">16</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/transformation1-1.png" width="480" style="display: block; margin: auto;" /></p>
<p>Above you can see a nice example of heteroscedasticity in the data, illustrating the dependence of the variance on the mean count value. Instead, <code>DESeq2</code> uses a regularized log transform (rlog) of the normalized counts for sample-level QC as it moderates the variance across the mean, improving the clustering.</p>
<p>Perform rlog-transformation:</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="cross.html#cb17-1" tabindex="-1"></a>rld <span class="ot">&lt;-</span> <span class="fu">rlog</span>(dds, <span class="at">blind =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<p>And this is how data looks like after rlog transformation:</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="cross.html#cb18-1" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="fu">as.data.frame</span>(<span class="fu">assay</span>(rld)), <span class="fu">aes</span>(<span class="at">x =</span> KO_PBS_642, <span class="at">y =</span> KO_PBS_641)) <span class="sc">+</span> </span>
<span id="cb18-2"><a href="cross.html#cb18-2" tabindex="-1"></a>            <span class="fu">geom_point</span>()<span class="sc">+</span></span>
<span id="cb18-3"><a href="cross.html#cb18-3" tabindex="-1"></a>            <span class="fu">ggtitle</span>(<span class="st">&#39;rlog transformation&#39;</span>)<span class="sc">+</span></span>
<span id="cb18-4"><a href="cross.html#cb18-4" tabindex="-1"></a>            <span class="fu">theme_dose</span>(<span class="dv">16</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/transformation_rlog-1.png" width="480" style="display: block; margin: auto;" /></p>
<div id="principal-component-analysis-pca" class="section level3 hasAnchor" number="3.2.1">
<h3><span class="header-section-number">3.2.1</span> Principal Component Analysis (PCA)<a href="cross.html#principal-component-analysis-pca" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>PCA is a technique used to emphasize variation and bring out strong patterns in a dataset (dimensionality reduction).</p>
<p>We would expect that biological replicates cluster together on PC1 and/or PC2, and the samples from different treatment groups cluster separately.</p>
<p>DESeq2 has a built-in function for plotting PCA plots, that uses ggplot2. It takes the rlog object as an input directly.</p>
<p>The function <code>plotPCA</code> requires two arguments as input: an rlog object and the intgroup (the column in our metadata that we are interested in).</p>
<p>Below we will create a custom function to plot PCA:</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="cross.html#cb19-1" tabindex="-1"></a>plotPCA.mystyle <span class="ot">&lt;-</span> <span class="cf">function</span> (object, <span class="at">ntop =</span> <span class="dv">500</span>, <span class="at">returnData =</span> <span class="cn">FALSE</span>)</span>
<span id="cb19-2"><a href="cross.html#cb19-2" tabindex="-1"></a>{</span>
<span id="cb19-3"><a href="cross.html#cb19-3" tabindex="-1"></a>    font.size <span class="ot">&lt;-</span> <span class="dv">18</span></span>
<span id="cb19-4"><a href="cross.html#cb19-4" tabindex="-1"></a>    rv <span class="ot">&lt;-</span> <span class="fu">rowVars</span>(<span class="fu">assay</span>(object))</span>
<span id="cb19-5"><a href="cross.html#cb19-5" tabindex="-1"></a>    r <span class="ot">&lt;-</span> <span class="fu">assay</span>(object)</span>
<span id="cb19-6"><a href="cross.html#cb19-6" tabindex="-1"></a>    select <span class="ot">&lt;-</span> <span class="fu">order</span>(rv, <span class="at">decreasing =</span> <span class="cn">TRUE</span>)[<span class="fu">seq_len</span>(<span class="fu">min</span>(ntop,</span>
<span id="cb19-7"><a href="cross.html#cb19-7" tabindex="-1"></a>        <span class="fu">length</span>(rv)))]</span>
<span id="cb19-8"><a href="cross.html#cb19-8" tabindex="-1"></a>    pca <span class="ot">&lt;-</span> <span class="fu">prcomp</span>(<span class="fu">t</span>(<span class="fu">assay</span>(object)[select, ]))</span>
<span id="cb19-9"><a href="cross.html#cb19-9" tabindex="-1"></a>    percentVar <span class="ot">&lt;-</span> pca<span class="sc">$</span>sdev<span class="sc">^</span><span class="dv">2</span><span class="sc">/</span><span class="fu">sum</span>(pca<span class="sc">$</span>sdev<span class="sc">^</span><span class="dv">2</span>)</span>
<span id="cb19-10"><a href="cross.html#cb19-10" tabindex="-1"></a>    </span>
<span id="cb19-11"><a href="cross.html#cb19-11" tabindex="-1"></a>    d1 <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">PC1 =</span> pca<span class="sc">$</span>x[, <span class="dv">1</span>], <span class="at">PC2 =</span> pca<span class="sc">$</span>x[, <span class="dv">2</span>], </span>
<span id="cb19-12"><a href="cross.html#cb19-12" tabindex="-1"></a>                     <span class="at">Genotype =</span> meta<span class="sc">$</span>Genotype, </span>
<span id="cb19-13"><a href="cross.html#cb19-13" tabindex="-1"></a>                     <span class="at">Treatment =</span> meta<span class="sc">$</span>Treatment,  </span>
<span id="cb19-14"><a href="cross.html#cb19-14" tabindex="-1"></a>                     <span class="at">name =</span> <span class="fu">colnames</span>(object))</span>
<span id="cb19-15"><a href="cross.html#cb19-15" tabindex="-1"></a>    </span>
<span id="cb19-16"><a href="cross.html#cb19-16" tabindex="-1"></a>    <span class="fu">ggplot</span>(<span class="at">data =</span> d1, <span class="fu">aes_string</span>(<span class="at">x =</span> <span class="st">&quot;PC1&quot;</span>, <span class="at">y =</span> <span class="st">&quot;PC2&quot;</span>)) <span class="sc">+</span></span>
<span id="cb19-17"><a href="cross.html#cb19-17" tabindex="-1"></a>        <span class="fu">geom_point</span>(<span class="fu">aes_string</span>(<span class="at">color =</span> <span class="st">&quot;Genotype&quot;</span>, <span class="at">shape =</span> <span class="st">&quot;Treatment&quot;</span>), <span class="at">size =</span> <span class="dv">6</span>) <span class="sc">+</span> </span>
<span id="cb19-18"><a href="cross.html#cb19-18" tabindex="-1"></a>        <span class="fu">xlab</span>(<span class="fu">paste0</span>(<span class="st">&quot;PC1: &quot;</span>, <span class="fu">round</span>(percentVar[<span class="dv">1</span>] <span class="sc">*</span> <span class="dv">100</span>), <span class="st">&quot;% variance&quot;</span>)) <span class="sc">+</span> </span>
<span id="cb19-19"><a href="cross.html#cb19-19" tabindex="-1"></a>        <span class="fu">ylab</span>(<span class="fu">paste0</span>(<span class="st">&quot;PC2: &quot;</span>, <span class="fu">round</span>(percentVar[<span class="dv">2</span>] <span class="sc">*</span> <span class="dv">100</span>), <span class="st">&quot;% variance&quot;</span>)) <span class="sc">+</span></span>
<span id="cb19-20"><a href="cross.html#cb19-20" tabindex="-1"></a>        <span class="fu">theme_dose</span>(<span class="at">font.size =</span> font.size)<span class="sc">+</span> <span class="co">#+ geom_label_repel(aes(label = colnames(data)), label.size = 0.1, box.padding = 0.2)</span></span>
<span id="cb19-21"><a href="cross.html#cb19-21" tabindex="-1"></a>        <span class="fu">theme</span>(</span>
<span id="cb19-22"><a href="cross.html#cb19-22" tabindex="-1"></a>              <span class="at">legend.key =</span> <span class="fu">element_rect</span>(<span class="at">colour =</span> <span class="cn">NA</span>, <span class="at">fill =</span> <span class="cn">NA</span>), </span>
<span id="cb19-23"><a href="cross.html#cb19-23" tabindex="-1"></a>              <span class="at">legend.title=</span> <span class="fu">element_blank</span>(), </span>
<span id="cb19-24"><a href="cross.html#cb19-24" tabindex="-1"></a>              <span class="at">legend.text=</span><span class="fu">element_text</span>(<span class="at">size=</span>font.size<span class="dv">-2</span>)</span>
<span id="cb19-25"><a href="cross.html#cb19-25" tabindex="-1"></a>        )</span>
<span id="cb19-26"><a href="cross.html#cb19-26" tabindex="-1"></a>} </span></code></pre></div>
<p>Plot PCA graph:</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="cross.html#cb20-1" tabindex="-1"></a><span class="fu">plotPCA.mystyle</span>(rld)</span></code></pre></div>
<p><img src="_main_files/figure-html/pca-1.png" width="624" style="display: block; margin: auto;" /></p>
<p>By default the function <code>plotPCA.custom</code> uses the top 500 most variable genes. You can change this by adding the ntop argument and specifying how many genes you want to use to draw the plot.</p>
<div class="design">
<p><strong>Task 1</strong>: In the example above, we just plotted the graph for PC1 and PC2 values. Please change the function <code>plotPCA.mystyle</code> so it can produce score plots for PC1 and PC3. Also, provide an obtained plot and a brief explanation of it. (1 point).</p>
</div>
</div>
<div id="correlation-heatmap" class="section level3 hasAnchor" number="3.2.2">
<h3><span class="header-section-number">3.2.2</span> Correlation Heatmap<a href="cross.html#correlation-heatmap" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The heatmap displays the correlation of gene expression for all pairwise combinations of samples in the dataset. Since the majority of genes are not differentially expressed, samples generally have high correlations with each other (values higher than 0.80). Samples below 0.80 may indicate an outlier in your data and/or sample contamination.</p>
<p>Extract the rlog matrix from the object:</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="cross.html#cb21-1" tabindex="-1"></a>rld_mat <span class="ot">&lt;-</span> <span class="fu">assay</span>(rld)</span></code></pre></div>
<p>Since there is no built-in function for heatmaps in <code>DESeq2</code> we will be using the <code>pheatmap()</code> function from the <code>pheatmap</code> package. This function requires a matrix/dataframe of numeric values as input, and so the first thing we need to is retrieve that information from the rld object:</p>
<p>Then we need to compute the pairwise correlation values for samples. We can do this using the cor() function:</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="cross.html#cb22-1" tabindex="-1"></a>rld_cor <span class="ot">&lt;-</span> <span class="fu">cor</span>(rld_mat)</span></code></pre></div>
<p>Plot the correlation values as a heatmap:</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="cross.html#cb23-1" tabindex="-1"></a><span class="fu">pheatmap</span>(rld_cor, <span class="at">annotation =</span> meta[,<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>)])</span></code></pre></div>
<p><img src="_main_files/figure-html/heatmap1-1.png" width="672" style="display: block; margin: auto;" /></p>

</div>
</div>
</div>
<div class="footnotes">
<hr />
<ol start="1">
<li id="fn1"><p>If you forgot how this method works you may check the lecture slides or watch this video: <iframe width="560" height="315" src="https://www.youtube.com/embed/UFB993xufUU" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe><a href="cross.html#fnref1" class="footnote-back">↩︎</a></p></li>
</ol>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="input-dataset.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="differential-testing.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/USERNAME/REPO/edit/BRANCH/02-cross-refs.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["_main.pdf", "_main.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
