<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 2 Input dataset | Omics Data Analysis</title>
  <meta name="description" content="This is a collection of seminars and homeworks for the Omics Data Analysis course at Skoltech." />
  <meta name="generator" content="bookdown 0.37 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 2 Input dataset | Omics Data Analysis" />
  <meta property="og:type" content="book" />
  <meta property="og:image" content="/docs/images/course_logo_v2.png" />
  <meta property="og:description" content="This is a collection of seminars and homeworks for the Omics Data Analysis course at Skoltech." />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 2 Input dataset | Omics Data Analysis" />
  
  <meta name="twitter:description" content="This is a collection of seminars and homeworks for the Omics Data Analysis course at Skoltech." />
  <meta name="twitter:image" content="/docs/images/course_logo_v2.png" />

<meta name="author" content="Dmitrii Smirnov, Skolkovo Institute of Science and Technology" />


<meta name="date" content="2024-03-27" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="index.html"/>
<link rel="next" href="cross.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<link href="libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet" />
<script src="libs/pagedtable-1.1/js/pagedtable.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="toc-logo"><a href="./"><img src="images/Logo.png"></a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#how-to-deal-with-r"><i class="fa fa-check"></i><b>1.1</b> How to deal with R</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#assignment-and-grading"><i class="fa fa-check"></i><b>1.2</b> Assignment and grading</a></li>
</ul></li>
<li class="part"><span><b>Lecture 4: DE analysis</b></span></li>
<li class="chapter" data-level="2" data-path="input-dataset.html"><a href="input-dataset.html"><i class="fa fa-check"></i><b>2</b> Input dataset</a>
<ul>
<li class="chapter" data-level="2.1" data-path="input-dataset.html"><a href="input-dataset.html#count-table-importing"><i class="fa fa-check"></i><b>2.1</b> Count table importing</a></li>
<li class="chapter" data-level="2.2" data-path="input-dataset.html"><a href="input-dataset.html#deseq2-object"><i class="fa fa-check"></i><b>2.2</b> DESeq2 object</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="cross.html"><a href="cross.html"><i class="fa fa-check"></i><b>3</b> Normalization and Quality control</a>
<ul>
<li class="chapter" data-level="3.1" data-path="cross.html"><a href="cross.html#normalization"><i class="fa fa-check"></i><b>3.1</b> Normalization</a></li>
<li class="chapter" data-level="3.2" data-path="cross.html"><a href="cross.html#quality-control"><i class="fa fa-check"></i><b>3.2</b> Quality control</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="cross.html"><a href="cross.html#principal-component-analysis-pca"><i class="fa fa-check"></i><b>3.2.1</b> Principal Component Analysis (PCA)</a></li>
<li class="chapter" data-level="3.2.2" data-path="cross.html"><a href="cross.html#correlation-heatmap"><i class="fa fa-check"></i><b>3.2.2</b> Correlation Heatmap</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="differential-testing.html"><a href="differential-testing.html"><i class="fa fa-check"></i><b>4</b> Differential testing</a>
<ul>
<li class="chapter" data-level="4.1" data-path="differential-testing.html"><a href="differential-testing.html#de-analysis"><i class="fa fa-check"></i><b>4.1</b> DE analysis</a></li>
<li class="chapter" data-level="4.2" data-path="differential-testing.html"><a href="differential-testing.html#extracting-significant-genes"><i class="fa fa-check"></i><b>4.2</b> Extracting significant genes</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="gene-annotation.html"><a href="gene-annotation.html"><i class="fa fa-check"></i><b>5</b> Gene annotation</a></li>
<li class="chapter" data-level="6" data-path="visualization.html"><a href="visualization.html"><i class="fa fa-check"></i><b>6</b> Visualization</a>
<ul>
<li class="chapter" data-level="6.1" data-path="visualization.html"><a href="visualization.html#volcano-plot"><i class="fa fa-check"></i><b>6.1</b> Volcano plot</a></li>
<li class="chapter" data-level="6.2" data-path="visualization.html"><a href="visualization.html#heatmap-of-significant-genes"><i class="fa fa-check"></i><b>6.2</b> Heatmap of significant genes</a></li>
<li class="chapter" data-level="6.3" data-path="visualization.html"><a href="visualization.html#comparison-with-dam-markers"><i class="fa fa-check"></i><b>6.3</b> Comparison with DAM markers</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="functional-analysis.html"><a href="functional-analysis.html"><i class="fa fa-check"></i><b>7</b> Functional analysis</a>
<ul>
<li class="chapter" data-level="7.1" data-path="functional-analysis.html"><a href="functional-analysis.html#gene-ontology-go-analysis"><i class="fa fa-check"></i><b>7.1</b> Gene Ontology (GO) analysis</a></li>
<li class="chapter" data-level="7.2" data-path="functional-analysis.html"><a href="functional-analysis.html#pathway-analysis"><i class="fa fa-check"></i><b>7.2</b> Pathway analysis</a></li>
<li class="chapter" data-level="7.3" data-path="functional-analysis.html"><a href="functional-analysis.html#gene-set-enrichment-analysis-gsea"><i class="fa fa-check"></i><b>7.3</b> Gene-Set Enrichment Analysis (GSEA)</a></li>
<li class="chapter" data-level="7.4" data-path="functional-analysis.html"><a href="functional-analysis.html#gene-set-enrichment-analysis-with-msigdb"><i class="fa fa-check"></i><b>7.4</b> Gene Set Enrichment Analysis with mSigDB</a></li>
</ul></li>
<li class="part"><span><b>Lecture 9: Transcriptome</b></span></li>
<li class="chapter" data-level="8" data-path="introduction-1.html"><a href="introduction-1.html"><i class="fa fa-check"></i><b>8</b> Introduction</a>
<ul>
<li class="chapter" data-level="8.1" data-path="introduction-1.html"><a href="introduction-1.html#assignment-and-grading-1"><i class="fa fa-check"></i><b>8.1</b> Assignment and grading</a></li>
<li class="chapter" data-level="8.2" data-path="introduction-1.html"><a href="introduction-1.html#set-up-conda-environment"><i class="fa fa-check"></i><b>8.2</b> Set up conda environment</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="de-novo-transcriptome-assembly.html"><a href="de-novo-transcriptome-assembly.html"><i class="fa fa-check"></i><b>9</b> De novo transcriptome assembly</a>
<ul>
<li class="chapter" data-level="9.1" data-path="de-novo-transcriptome-assembly.html"><a href="de-novo-transcriptome-assembly.html#read-filtering"><i class="fa fa-check"></i><b>9.1</b> Read filtering</a></li>
<li class="chapter" data-level="9.2" data-path="de-novo-transcriptome-assembly.html"><a href="de-novo-transcriptome-assembly.html#run-trinity"><i class="fa fa-check"></i><b>9.2</b> Run trinity</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="transcriptome-assembly-qc.html"><a href="transcriptome-assembly-qc.html"><i class="fa fa-check"></i><b>10</b> Transcriptome assembly QC</a>
<ul>
<li class="chapter" data-level="10.1" data-path="transcriptome-assembly-qc.html"><a href="transcriptome-assembly-qc.html#obtain-assembly-statistics"><i class="fa fa-check"></i><b>10.1</b> Obtain assembly statistics</a></li>
<li class="chapter" data-level="10.2" data-path="transcriptome-assembly-qc.html"><a href="transcriptome-assembly-qc.html#transcript-filtering"><i class="fa fa-check"></i><b>10.2</b> Transcript filtering</a></li>
<li class="chapter" data-level="10.3" data-path="transcriptome-assembly-qc.html"><a href="transcriptome-assembly-qc.html#assembly-completeness-with-gvolante"><i class="fa fa-check"></i><b>10.3</b> Assembly completeness with gVolante</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="quantification-and-annotation.html"><a href="quantification-and-annotation.html"><i class="fa fa-check"></i><b>11</b> Quantification and annotation</a>
<ul>
<li class="chapter" data-level="11.1" data-path="quantification-and-annotation.html"><a href="quantification-and-annotation.html#transcript-quantification"><i class="fa fa-check"></i><b>11.1</b> Transcript quantification</a></li>
<li class="chapter" data-level="11.2" data-path="quantification-and-annotation.html"><a href="quantification-and-annotation.html#transcript-annotation"><i class="fa fa-check"></i><b>11.2</b> Transcript annotation</a></li>
<li class="chapter" data-level="11.3" data-path="quantification-and-annotation.html"><a href="quantification-and-annotation.html#integrating-expression-data-and-annotation"><i class="fa fa-check"></i><b>11.3</b> Integrating expression data and annotation</a></li>
</ul></li>
<li class="part"><span><b>Lecture 13: Lipidomics</b></span></li>
<li class="chapter" data-level="12" data-path="lipidomics-data-analysis.html"><a href="lipidomics-data-analysis.html"><i class="fa fa-check"></i><b>12</b> Lipidomics Data Analysis</a></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="part"><span><b>Session info</b></span></li>
<li class="chapter" data-level="13" data-path="session-info.html"><a href="session-info.html"><i class="fa fa-check"></i><b>13</b> Session Info</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Omics Data Analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="input-dataset" class="section level1 hasAnchor" number="2">
<h1><span class="header-section-number">Chapter 2</span> Input dataset<a href="input-dataset.html#input-dataset" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>Today we will focus on the differential gene expression analysis to study the gene expression signatures of microglia from accelerated aging mice subjected to PBS (phosphate buffered saline) and LPS (lipopolysaccharides, activated microglia) treatments from Zhang et al., 2022 (<a href="https://doi.org/10.1186/s12974-022-02463-5" class="uri">https://doi.org/10.1186/s12974-022-02463-5</a>). Our dataset includes 2 experimental groups of replicates: Control and KO microglia, where KO group corresponds to to the model of accelerated aging, caused by deficiency of the DNA-repair associated gene Ercc1. One half of the samples in each group we additionally treated with PBS and another half were treated with LPS to achieve pro-inflammatory microglial activation. Raw reads were already preprocessed by the authors of the original study, therefore we can skip a large part devoted to raw read processing and mapping (Figure 1), that you already studied during the previous lecture and start with a table of counts. The input dataset is accessible here:</p>
<ul>
<li>Course server: <code>/home/d.smirnov/homework4/Data/Data.csv</code></li>
<li>Yandex disk: <a href="https://disk.yandex.ru/d/2X3QgYRa5iqcFw" class="uri">https://disk.yandex.ru/d/2X3QgYRa5iqcFw</a></li>
</ul>
<p><img src="docs/images/TitleFigure.png"/> <strong>Figure 1.</strong> DE analysis typical workflow</p>
<p>Import libraries:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="input-dataset.html#cb1-1" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb1-2"><a href="input-dataset.html#cb1-2" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb1-3"><a href="input-dataset.html#cb1-3" tabindex="-1"></a><span class="fu">library</span>(DESeq2)</span>
<span id="cb1-4"><a href="input-dataset.html#cb1-4" tabindex="-1"></a><span class="fu">library</span>(DOSE)</span>
<span id="cb1-5"><a href="input-dataset.html#cb1-5" tabindex="-1"></a><span class="fu">library</span>(pheatmap)</span>
<span id="cb1-6"><a href="input-dataset.html#cb1-6" tabindex="-1"></a><span class="fu">library</span>(tibble)</span>
<span id="cb1-7"><a href="input-dataset.html#cb1-7" tabindex="-1"></a><span class="fu">library</span>(clusterProfiler)</span></code></pre></div>
<div id="count-table-importing" class="section level2 hasAnchor" number="2.1">
<h2><span class="header-section-number">2.1</span> Count table importing<a href="input-dataset.html#count-table-importing" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We will start the analysis with uploading <strong>raw</strong> count data to the R workspace.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="input-dataset.html#cb2-1" tabindex="-1"></a>data <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;data/Data.csv&quot;</span>, <span class="at">row.names =</span> <span class="dv">1</span>)</span></code></pre></div>
<p>The gene expression table has gene names in the rows and samples names in the columns. ‘Length’ column denotes a length of each annotated gene:</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="input-dataset.html#cb3-1" tabindex="-1"></a><span class="fu">paged_table</span>(<span class="fu">head</span>(data))</span></code></pre></div>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":[""],"name":["_rn_"],"type":[""],"align":["left"]},{"label":["Length"],"name":[1],"type":["int"],"align":["right"]},{"label":["KO_PBS_642"],"name":[2],"type":["dbl"],"align":["right"]},{"label":["KO_PBS_641"],"name":[3],"type":["dbl"],"align":["right"]},{"label":["KO_PBS_638"],"name":[4],"type":["dbl"],"align":["right"]},{"label":["KO_LPS_600"],"name":[5],"type":["dbl"],"align":["right"]},{"label":["KO_LPS_194"],"name":[6],"type":["dbl"],"align":["right"]},{"label":["KO_LPS_193"],"name":[7],"type":["dbl"],"align":["right"]},{"label":["Control_PBS_175"],"name":[8],"type":["dbl"],"align":["right"]},{"label":["Control_PBS_171"],"name":[9],"type":["dbl"],"align":["right"]},{"label":["Control_PBS_170"],"name":[10],"type":["dbl"],"align":["right"]},{"label":["Control_LPS_185"],"name":[11],"type":["dbl"],"align":["right"]},{"label":["Control_LPS_184"],"name":[12],"type":["dbl"],"align":["right"]},{"label":["Control_LPS_149"],"name":[13],"type":["dbl"],"align":["right"]}],"data":[{"1":"1070","2":"0","3":"0","4":"0","5":"0","6":"0","7":"0","8":"0","9":"0","10":"0","11":"0","12":"0","13":"0","_rn_":"ENSMUSG00000102693"},{"1":"110","2":"0","3":"0","4":"0","5":"0","6":"0","7":"0","8":"0","9":"0","10":"0","11":"0","12":"0","13":"0","_rn_":"ENSMUSG00000064842"},{"1":"6094","2":"4","3":"0","4":"0","5":"0","6":"0","7":"0","8":"0","9":"0","10":"0","11":"1","12":"10","13":"12","_rn_":"ENSMUSG00000051951"},{"1":"480","2":"0","3":"0","4":"0","5":"0","6":"0","7":"0","8":"0","9":"0","10":"0","11":"0","12":"0","13":"0","_rn_":"ENSMUSG00000102851"},{"1":"2819","2":"0","3":"0","4":"0","5":"4","6":"0","7":"0","8":"0","9":"0","10":"0","11":"12","12":"0","13":"2","_rn_":"ENSMUSG00000103377"},{"1":"2233","2":"0","3":"0","4":"0","5":"0","6":"1","7":"0","8":"0","9":"0","10":"0","11":"0","12":"0","13":"5","_rn_":"ENSMUSG00000104017"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<p>Further we will need a meta table which maps our samples to the corresponding sample groups that we are investigating. Our metadata will include three columns of interest: “Genotype”, “Treatment” and “SampleID”.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="input-dataset.html#cb4-1" tabindex="-1"></a>meta <span class="ot">&lt;-</span> <span class="fu">colnames</span>(data) <span class="sc">%&gt;%</span> </span>
<span id="cb4-2"><a href="input-dataset.html#cb4-2" tabindex="-1"></a>        .[<span class="sc">-</span><span class="fu">c</span>(<span class="dv">1</span>)] <span class="sc">%&gt;%</span> <span class="fu">strsplit</span>(<span class="at">split =</span> <span class="st">&#39;_&#39;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb4-3"><a href="input-dataset.html#cb4-3" tabindex="-1"></a>        <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb4-4"><a href="input-dataset.html#cb4-4" tabindex="-1"></a>        <span class="fu">t</span>() </span>
<span id="cb4-5"><a href="input-dataset.html#cb4-5" tabindex="-1"></a><span class="fu">rownames</span>(meta) <span class="ot">&lt;-</span> <span class="fu">colnames</span>(data)[<span class="sc">-</span><span class="fu">c</span>(<span class="dv">1</span>)]</span>
<span id="cb4-6"><a href="input-dataset.html#cb4-6" tabindex="-1"></a><span class="fu">colnames</span>(meta) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;Genotype&quot;</span>, <span class="st">&quot;Treatment&quot;</span>, <span class="st">&quot;SampleID&quot;</span>)</span>
<span id="cb4-7"><a href="input-dataset.html#cb4-7" tabindex="-1"></a>meta <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(meta)</span></code></pre></div>
<p>Check the generated meta table:</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="input-dataset.html#cb5-1" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(meta)</span></code></pre></div>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="left">Genotype</th>
<th align="left">Treatment</th>
<th align="left">SampleID</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">KO_PBS_642</td>
<td align="left">KO</td>
<td align="left">PBS</td>
<td align="left">642</td>
</tr>
<tr class="even">
<td align="left">KO_PBS_641</td>
<td align="left">KO</td>
<td align="left">PBS</td>
<td align="left">641</td>
</tr>
<tr class="odd">
<td align="left">KO_PBS_638</td>
<td align="left">KO</td>
<td align="left">PBS</td>
<td align="left">638</td>
</tr>
<tr class="even">
<td align="left">KO_LPS_600</td>
<td align="left">KO</td>
<td align="left">LPS</td>
<td align="left">600</td>
</tr>
<tr class="odd">
<td align="left">KO_LPS_194</td>
<td align="left">KO</td>
<td align="left">LPS</td>
<td align="left">194</td>
</tr>
<tr class="even">
<td align="left">KO_LPS_193</td>
<td align="left">KO</td>
<td align="left">LPS</td>
<td align="left">193</td>
</tr>
<tr class="odd">
<td align="left">Control_PBS_175</td>
<td align="left">Control</td>
<td align="left">PBS</td>
<td align="left">175</td>
</tr>
<tr class="even">
<td align="left">Control_PBS_171</td>
<td align="left">Control</td>
<td align="left">PBS</td>
<td align="left">171</td>
</tr>
<tr class="odd">
<td align="left">Control_PBS_170</td>
<td align="left">Control</td>
<td align="left">PBS</td>
<td align="left">170</td>
</tr>
<tr class="even">
<td align="left">Control_LPS_185</td>
<td align="left">Control</td>
<td align="left">LPS</td>
<td align="left">185</td>
</tr>
<tr class="odd">
<td align="left">Control_LPS_184</td>
<td align="left">Control</td>
<td align="left">LPS</td>
<td align="left">184</td>
</tr>
<tr class="even">
<td align="left">Control_LPS_149</td>
<td align="left">Control</td>
<td align="left">LPS</td>
<td align="left">149</td>
</tr>
</tbody>
</table>
<p>Another important step is a filtering of the low- and non-expressed genes prior statistical analysis. Such genes contributes to the False Discovery Rate. We will keep only those genes having at least 10 counts in total across experimental replicates.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="input-dataset.html#cb6-1" tabindex="-1"></a>data <span class="ot">&lt;-</span> data[<span class="fu">rowSums</span>(data[,<span class="sc">-</span><span class="fu">c</span>(<span class="dv">1</span>)]) <span class="sc">&gt;</span> <span class="dv">10</span>, ]</span></code></pre></div>
</div>
<div id="deseq2-object" class="section level2 hasAnchor" number="2.2">
<h2><span class="header-section-number">2.2</span> DESeq2 object<a href="input-dataset.html#deseq2-object" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Let’s start the main part of the analysis with creating the <code>DESeqDataSet</code> object. To create this object we will need the count matrix and the metadata table as input. We will also need to specify a design formula. The design formula specifies the column(s) in the metadata table and how they should be used in the analysis. For our dataset we only have two column we are interested in, that is <span class="math inline">\(\sim Treatment + Genotype\)</span>. The <span class="math inline">\(Genotype\)</span> is the main factor of interest, therefore it is in the end of the design formula.</p>
<p>Create DESeq2Dataset object:</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="input-dataset.html#cb7-1" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> <span class="fu">round</span>(data[,<span class="sc">-</span><span class="fu">c</span>(<span class="dv">1</span>)]), </span>
<span id="cb7-2"><a href="input-dataset.html#cb7-2" tabindex="-1"></a>                              <span class="at">colData =</span> meta, </span>
<span id="cb7-3"><a href="input-dataset.html#cb7-3" tabindex="-1"></a>                              <span class="at">design =</span> <span class="sc">~</span> Treatment <span class="sc">+</span> Genotype)</span></code></pre></div>
<pre><code>## converting counts to integer mode</code></pre>
<pre><code>## Warning in DESeqDataSet(se, design = design, ignoreRank): some variables in design formula are characters, converting to factors</code></pre>
<p>Check what is inside the object:</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="input-dataset.html#cb10-1" tabindex="-1"></a>dds</span></code></pre></div>
<pre><code>## class: DESeqDataSet 
## dim: 22783 12 
## metadata(1): version
## assays(1): counts
## rownames(22783): ENSMUSG00000051951 ENSMUSG00000103377 ... ENSMUSG00000063897 ENSMUSG00000095742
## rowData names(0):
## colnames(12): KO_PBS_642 KO_PBS_641 ... Control_LPS_184 Control_LPS_149
## colData names(3): Genotype Treatment SampleID</code></pre>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="index.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="cross.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/USERNAME/REPO/edit/BRANCH/01-import.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["_main.pdf", "_main.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
