<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 13 Session Info | Omics Data Analysis</title>
  <meta name="description" content="This is a collection of seminars and homeworks for the Omics Data Analysis course at Skoltech." />
  <meta name="generator" content="bookdown 0.37 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 13 Session Info | Omics Data Analysis" />
  <meta property="og:type" content="book" />
  <meta property="og:image" content="/docs/images/course_logo_v2.png" />
  <meta property="og:description" content="This is a collection of seminars and homeworks for the Omics Data Analysis course at Skoltech." />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 13 Session Info | Omics Data Analysis" />
  
  <meta name="twitter:description" content="This is a collection of seminars and homeworks for the Omics Data Analysis course at Skoltech." />
  <meta name="twitter:image" content="/docs/images/course_logo_v2.png" />

<meta name="author" content="Dmitrii Smirnov, Skolkovo Institute of Science and Technology" />


<meta name="date" content="2024-03-27" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="references.html"/>

<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<link href="libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet" />
<script src="libs/pagedtable-1.1/js/pagedtable.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="toc-logo"><a href="./"><img src="images/Logo.png"></a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#how-to-deal-with-r"><i class="fa fa-check"></i><b>1.1</b> How to deal with R</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#assignment-and-grading"><i class="fa fa-check"></i><b>1.2</b> Assignment and grading</a></li>
</ul></li>
<li class="part"><span><b>Lecture 4: DE analysis</b></span></li>
<li class="chapter" data-level="2" data-path="input-dataset.html"><a href="input-dataset.html"><i class="fa fa-check"></i><b>2</b> Input dataset</a>
<ul>
<li class="chapter" data-level="2.1" data-path="input-dataset.html"><a href="input-dataset.html#count-table-importing"><i class="fa fa-check"></i><b>2.1</b> Count table importing</a></li>
<li class="chapter" data-level="2.2" data-path="input-dataset.html"><a href="input-dataset.html#deseq2-object"><i class="fa fa-check"></i><b>2.2</b> DESeq2 object</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="cross.html"><a href="cross.html"><i class="fa fa-check"></i><b>3</b> Normalization and Quality control</a>
<ul>
<li class="chapter" data-level="3.1" data-path="cross.html"><a href="cross.html#normalization"><i class="fa fa-check"></i><b>3.1</b> Normalization</a></li>
<li class="chapter" data-level="3.2" data-path="cross.html"><a href="cross.html#quality-control"><i class="fa fa-check"></i><b>3.2</b> Quality control</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="cross.html"><a href="cross.html#principal-component-analysis-pca"><i class="fa fa-check"></i><b>3.2.1</b> Principal Component Analysis (PCA)</a></li>
<li class="chapter" data-level="3.2.2" data-path="cross.html"><a href="cross.html#correlation-heatmap"><i class="fa fa-check"></i><b>3.2.2</b> Correlation Heatmap</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="differential-testing.html"><a href="differential-testing.html"><i class="fa fa-check"></i><b>4</b> Differential testing</a>
<ul>
<li class="chapter" data-level="4.1" data-path="differential-testing.html"><a href="differential-testing.html#de-analysis"><i class="fa fa-check"></i><b>4.1</b> DE analysis</a></li>
<li class="chapter" data-level="4.2" data-path="differential-testing.html"><a href="differential-testing.html#extracting-significant-genes"><i class="fa fa-check"></i><b>4.2</b> Extracting significant genes</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="gene-annotation.html"><a href="gene-annotation.html"><i class="fa fa-check"></i><b>5</b> Gene annotation</a></li>
<li class="chapter" data-level="6" data-path="visualization.html"><a href="visualization.html"><i class="fa fa-check"></i><b>6</b> Visualization</a>
<ul>
<li class="chapter" data-level="6.1" data-path="visualization.html"><a href="visualization.html#volcano-plot"><i class="fa fa-check"></i><b>6.1</b> Volcano plot</a></li>
<li class="chapter" data-level="6.2" data-path="visualization.html"><a href="visualization.html#heatmap-of-significant-genes"><i class="fa fa-check"></i><b>6.2</b> Heatmap of significant genes</a></li>
<li class="chapter" data-level="6.3" data-path="visualization.html"><a href="visualization.html#comparison-with-dam-markers"><i class="fa fa-check"></i><b>6.3</b> Comparison with DAM markers</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="functional-analysis.html"><a href="functional-analysis.html"><i class="fa fa-check"></i><b>7</b> Functional analysis</a>
<ul>
<li class="chapter" data-level="7.1" data-path="functional-analysis.html"><a href="functional-analysis.html#gene-ontology-go-analysis"><i class="fa fa-check"></i><b>7.1</b> Gene Ontology (GO) analysis</a></li>
<li class="chapter" data-level="7.2" data-path="functional-analysis.html"><a href="functional-analysis.html#pathway-analysis"><i class="fa fa-check"></i><b>7.2</b> Pathway analysis</a></li>
<li class="chapter" data-level="7.3" data-path="functional-analysis.html"><a href="functional-analysis.html#gene-set-enrichment-analysis-gsea"><i class="fa fa-check"></i><b>7.3</b> Gene-Set Enrichment Analysis (GSEA)</a></li>
<li class="chapter" data-level="7.4" data-path="functional-analysis.html"><a href="functional-analysis.html#gene-set-enrichment-analysis-with-msigdb"><i class="fa fa-check"></i><b>7.4</b> Gene Set Enrichment Analysis with mSigDB</a></li>
</ul></li>
<li class="part"><span><b>Lecture 9: Transcriptome</b></span></li>
<li class="chapter" data-level="8" data-path="introduction-1.html"><a href="introduction-1.html"><i class="fa fa-check"></i><b>8</b> Introduction</a>
<ul>
<li class="chapter" data-level="8.1" data-path="introduction-1.html"><a href="introduction-1.html#assignment-and-grading-1"><i class="fa fa-check"></i><b>8.1</b> Assignment and grading</a></li>
<li class="chapter" data-level="8.2" data-path="introduction-1.html"><a href="introduction-1.html#set-up-conda-environment"><i class="fa fa-check"></i><b>8.2</b> Set up conda environment</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="de-novo-transcriptome-assembly.html"><a href="de-novo-transcriptome-assembly.html"><i class="fa fa-check"></i><b>9</b> De novo transcriptome assembly</a>
<ul>
<li class="chapter" data-level="9.1" data-path="de-novo-transcriptome-assembly.html"><a href="de-novo-transcriptome-assembly.html#read-filtering"><i class="fa fa-check"></i><b>9.1</b> Read filtering</a></li>
<li class="chapter" data-level="9.2" data-path="de-novo-transcriptome-assembly.html"><a href="de-novo-transcriptome-assembly.html#run-trinity"><i class="fa fa-check"></i><b>9.2</b> Run trinity</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="transcriptome-assembly-qc.html"><a href="transcriptome-assembly-qc.html"><i class="fa fa-check"></i><b>10</b> Transcriptome assembly QC</a>
<ul>
<li class="chapter" data-level="10.1" data-path="transcriptome-assembly-qc.html"><a href="transcriptome-assembly-qc.html#obtain-assembly-statistics"><i class="fa fa-check"></i><b>10.1</b> Obtain assembly statistics</a></li>
<li class="chapter" data-level="10.2" data-path="transcriptome-assembly-qc.html"><a href="transcriptome-assembly-qc.html#transcript-filtering"><i class="fa fa-check"></i><b>10.2</b> Transcript filtering</a></li>
<li class="chapter" data-level="10.3" data-path="transcriptome-assembly-qc.html"><a href="transcriptome-assembly-qc.html#assembly-completeness-with-gvolante"><i class="fa fa-check"></i><b>10.3</b> Assembly completeness with gVolante</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="quantification-and-annotation.html"><a href="quantification-and-annotation.html"><i class="fa fa-check"></i><b>11</b> Quantification and annotation</a>
<ul>
<li class="chapter" data-level="11.1" data-path="quantification-and-annotation.html"><a href="quantification-and-annotation.html#transcript-quantification"><i class="fa fa-check"></i><b>11.1</b> Transcript quantification</a></li>
<li class="chapter" data-level="11.2" data-path="quantification-and-annotation.html"><a href="quantification-and-annotation.html#transcript-annotation"><i class="fa fa-check"></i><b>11.2</b> Transcript annotation</a></li>
<li class="chapter" data-level="11.3" data-path="quantification-and-annotation.html"><a href="quantification-and-annotation.html#integrating-expression-data-and-annotation"><i class="fa fa-check"></i><b>11.3</b> Integrating expression data and annotation</a></li>
</ul></li>
<li class="part"><span><b>Lecture 13: Lipidomics</b></span></li>
<li class="chapter" data-level="12" data-path="lipidomics-data-analysis.html"><a href="lipidomics-data-analysis.html"><i class="fa fa-check"></i><b>12</b> Lipidomics Data Analysis</a></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="part"><span><b>Session info</b></span></li>
<li class="chapter" data-level="13" data-path="session-info.html"><a href="session-info.html"><i class="fa fa-check"></i><b>13</b> Session Info</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Omics Data Analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="session-info" class="section level1 hasAnchor" number="13">
<h1><span class="header-section-number">Chapter 13</span> Session Info<a href="session-info.html#session-info" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div class="sourceCode" id="cb147"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb147-1"><a href="session-info.html#cb147-1" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code></pre></div>
<pre><code>## R version 4.3.1 (2023-06-16)
## Platform: aarch64-apple-darwin20 (64-bit)
## Running under: macOS Ventura 13.6
## 
## Matrix products: default
## BLAS:   /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libBLAS.dylib 
## LAPACK: /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0
## 
## locale:
## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
## 
## time zone: Europe/Moscow
## tzcode source: internal
## 
## attached base packages:
## [1] stats4    stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
##  [1] msigdbr_7.5.1               enrichplot_1.20.3           EnhancedVolcano_1.18.0      ggrepel_0.9.5               org.Mm.eg.db_3.17.0        
##  [6] AnnotationDbi_1.62.2        clusterProfiler_4.8.3       tibble_3.2.1                pheatmap_1.0.12             DOSE_3.26.2                
## [11] DESeq2_1.40.2               SummarizedExperiment_1.30.2 Biobase_2.60.0              MatrixGenerics_1.12.3       matrixStats_1.2.0          
## [16] GenomicRanges_1.52.1        GenomeInfoDb_1.36.4         IRanges_2.34.1              S4Vectors_0.38.2            BiocGenerics_0.46.0        
## [21] ggplot2_3.4.4               dplyr_1.1.4                 rmarkdown_2.25              reactable_0.4.4            
## 
## loaded via a namespace (and not attached):
##   [1] RColorBrewer_1.1-3      rstudioapi_0.15.0       jsonlite_1.8.8          magrittr_2.0.3          farver_2.1.1            fs_1.6.3               
##   [7] zlibbioc_1.46.0         vctrs_0.6.5             memoise_2.0.1           RCurl_1.98-1.13         ggtree_3.8.2            htmltools_0.5.7        
##  [13] S4Arrays_1.0.6          gridGraphics_0.5-1      sass_0.4.8              bslib_0.6.1             htmlwidgets_1.6.4       plyr_1.8.9             
##  [19] cachem_1.0.8            igraph_2.0.2            lifecycle_1.0.4         pkgconfig_2.0.3         Matrix_1.6-4            R6_2.5.1               
##  [25] fastmap_1.1.1           gson_0.1.0              GenomeInfoDbData_1.2.10 digest_0.6.34           aplot_0.2.2             colorspace_2.1-0       
##  [31] patchwork_1.2.0         RSQLite_2.3.5           fansi_1.0.6             httr_1.4.7              polyclip_1.10-6         abind_1.4-5            
##  [37] compiler_4.3.1          bit64_4.0.5             withr_3.0.0             downloader_0.4          BiocParallel_1.34.2     viridis_0.6.5          
##  [43] DBI_1.2.0               ggupset_0.3.0           ggforce_0.4.1           MASS_7.3-60             DelayedArray_0.26.7     HDO.db_0.99.1          
##  [49] tools_4.3.1             ape_5.7-1               scatterpie_0.2.1        glue_1.7.0              nlme_3.1-164            GOSemSim_2.26.1        
##  [55] grid_4.3.1              shadowtext_0.1.3        reshape2_1.4.4          fgsea_1.26.0            generics_0.1.3          gtable_0.3.4           
##  [61] tidyr_1.3.1             data.table_1.15.0       tidygraph_1.2.3         utf8_1.2.4              XVector_0.40.0          pillar_1.9.0           
##  [67] stringr_1.5.1           babelgene_22.9          yulab.utils_0.1.1       splines_4.3.1           tweenr_2.0.2            treeio_1.24.3          
##  [73] lattice_0.22-5          bit_4.0.5               tidyselect_1.2.0        GO.db_3.17.0            locfit_1.5-9.8          Biostrings_2.68.1      
##  [79] knitr_1.45              gridExtra_2.3           bookdown_0.37           xfun_0.42               graphlayouts_1.0.2      stringi_1.8.3          
##  [85] lazyeval_0.2.2          ggfun_0.1.3             yaml_2.3.8              evaluate_0.23           codetools_0.2-19        ggraph_2.1.0           
##  [91] qvalue_2.32.0           ggplotify_0.1.2         cli_3.6.2               munsell_0.5.0           jquerylib_0.1.4         Rcpp_1.0.12            
##  [97] png_0.1-8               parallel_4.3.1          blob_1.2.4              bitops_1.0-7            viridisLite_0.4.2       tidytree_0.4.5         
## [103] scales_1.3.0            purrr_1.0.2             crayon_1.5.2            rlang_1.1.3             cowplot_1.1.3           fastmatch_1.1-4        
## [109] KEGGREST_1.40.1</code></pre>

</div>






            </section>

          </div>
        </div>
      </div>
<a href="references.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/USERNAME/REPO/edit/BRANCH/13-sessioninfo.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["_main.pdf", "_main.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
